src/test/controller/BaseController.ts

```diff
@@ -5,71 +5,97 @@ import { auth } from 'firebase-admin/lib/auth';
 const firebaseMock = require('firebase-mock');
-import { AxiosInstance } from 'axios';
+import axios, { AxiosInstance } from 'axios';
 import nodeGeocoder, { Geocoder } from 'node-geocoder';
+import AWSMock from 'mock-aws-s3';
+import { S3 } from 'aws-sdk';
+import { S3_BUCKET } from '../../main/service/ResourceCleanupService';
 
-/**
- * Creates an express app and registers all handlers and routes
- * * Used for integration tests that test controllers (supertest needs an instance of express app without the port binding)
- * */
-export const registerTestApplication = async (
-  redisClient: RedisClient,
-  firebaseAdmin: Auth,
-  geocoder: Geocoder,
-  axiosInstance?: AxiosInstance
-): Promise<Express> => {
-  const app = new App();
-  const expressApp = express();
-  await app.registerHandlersAndRoutes(
-    expressApp,
-    redisClient,
-    firebaseAdmin,
-    geocoder,
-    axiosInstance
-  );
-  return expressApp;
-};
+export class BaseController {
+  mockRedisClient: RedisClient;
+  mockFirebaseAdmin: Auth;
+  mockGeoCoder: Geocoder;
+  mockS3: S3;
+  axiosInstance: AxiosInstance;
 
-export const connectRedisClient = async (): Promise<RedisClient> => {
-  // see https://www.npmjs.com/package/redis-mock
-  return redisMock.createClient();
-};
+  constructor() {
+    this.mockRedisClient = BaseController.createRedisMock();
+    this.mockFirebaseAdmin = BaseController.createFirebaseMock();
+    this.mockGeoCoder = BaseController.createMockNodeGeocoder();
+    this.mockS3 = BaseController.createS3Mock();
+    this.axiosInstance = axios.create();
+  }
 
-export const createFirebaseMock = (): Auth => {
-  const mockAuth = new firebaseMock.MockAuthentication();
-  const mockDatabase = new firebaseMock.MockFirebase();
-  // const mockFirestore = new firebaseMock.MockFirestore();
-  const mockStorage = new firebaseMock.MockStorage();
-  // const mockMessaging = new firebaseMock.MockMessaging();
-  const mockSdk = new firebaseMock.MockFirebaseSdk(
-    // use null if your code does not use RTDB
-    (path: any) => {
-      return path ? mockDatabase.child(path) : mockDatabase;
-    },
-    // use null if your code does not use AUTHENTICATION
-    () => {
-      return mockAuth;
-    },
-    // use null if your code does not use FIRESTORE
-    // () => {
-    //   return mockFirestore;
-    // },
-    null,
-    // use null if your code does not use STORAGE
-    () => {
-      return mockStorage;
-    },
-    // use null if your code does not use MESSAGING
-    // () => {
-    //   return mockMessaging;
-    // }
-    null
-  );
-  return mockSdk.auth();
-};
+  /**
+   * Creates an express app and registers all handlers and routes
+   * * Used for integration tests that test controllers (supertest needs an instance of express app without the port binding)
+   * */
+  registerTestApplication = async (): Promise<Express> => {
+    const app = new App();
+    const expressApp = express();
+    await app.registerHandlersAndRoutes(
```

Comment:
Should we wrap these in try/catch?

Reply:
I think it's okay, not catching exceptions in tests is okay I think because we are leaving it to the testing framework to output that an unexpected error occured